### PR TITLE
refactor(Modal): Upgrade Modal component to Antd5

### LINKED ISSUE #31402: Ant Design 5: Upgrade Modal component
Please refer to SIP-139 https://github.com/apache/superset/issues/29268.

The `Modal` component should be upgraded to Ant Design version 5.

Similar work has been completed [here](https://github.com/apache/superset/pulls?q=is%3Apr+label%3Afrontend%3Arefactor%3Aantd5+is%3Aclosed).

When upgrading the component to Ant Design 5, please ensure to include:

- Proper RTL test coverage if not present yet.
- A Storybook file (if not present) or all necessary changes.
- Include all relevant component options in the Storybook.
- Clean up unnecessary custom styles.

When opening a PR:
- Name your PR refactor({Component}): Upgrade {component} to Ant Design 5
- Add screenshots of BEFORE/AFTER in your PR.
- Update the [project tracker](https://github.com/apache/superset/discussions/29293) when your PR is merged.
- Make sure you or a committer has labeled your PR with [frontend:refactor:antd5](https://github.com/apache/superset/pulls?q=is%3Apr+label%3Afrontend%3Arefactor%3Aantd5+is%3Aclosed).

### PR DESCRIPTION
**Pull request by alexandrusoare**
Fixes #31402

### SUMMARY
Upgrading the Modal component to AntDesign5
**Info** In order to upgrade it completely I also need the Button component to be upgraded to Antdv5 as there are a lot of footers with custom buttons

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before
image

After
image

Before
image

After
image

Before
image

After
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ const ErrorModal = styled(Modal)<{ level: ErrorLevel }>`
   color: ${({ level, theme }) => theme.colors[level].dark2};
   overflow-wrap: break-word;
 
-  .ant-modal-header {
+  .antd5-modal-header {
     background-color: ${({ level, theme }) => theme.colors[level].light2};
```
Comment (geido) [inline superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx:74]: Can this be centralized / removed / or moved to theme level?

File: superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ const ErrorModal = styled(Modal)<{ level: ErrorLevel }>`
   color: ${({ level, theme }) => theme.colors[level].dark2};
```
Comment (geido) [inline superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx:70]: Can this be centralized / removed / or moved to theme level?

File: superset-frontend/src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -106,30 +111,41 @@ export const StyledModal = styled(BaseModal)<StyledModalProps>`
...
     margin-bottom: ${({ theme }) => theme.gridUnit * 4}px;
     margin-top: ${({ theme }) => theme.gridUnit * 4}px;
+    padding: 0;
   }
 
-  .ant-modal-header {
+  .antd5-modal-header {
     flex: 0 0 auto;
-    background-color: ${({ theme }) => theme.colors.grayscale.light4};
     border-radius: ${({ theme }) => theme.borderRadius}px
```
Comment (geido) [inline superset-frontend/src/components/Modal/Modal.tsx:125]: Can this be centralized / removed / or moved to theme level?

File: superset-frontend/src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -106,30 +111,41 @@ export const StyledModal = styled(BaseModal)<StyledModalProps>`
...
       ${({ theme }) => theme.borderRadius}px 0 0;
-    padding-left: ${({ theme }) => theme.gridUnit * 4}px;
-    padding-right: ${({ theme }) => theme.gridUnit * 4}px;
+    padding: ${({ theme }) => theme.gridUnit * 4}px
+      ${({ theme }) => theme.gridUnit * 6}px;
+
+    .antd5-modal-title {
+      font-weight: ${({ theme }) => theme.typography.weights.medium};
```
Comment (geido) [inline superset-frontend/src/components/Modal/Modal.tsx:131]: Can this be centralized / removed / or moved to theme level?

File: superset-frontend/src/theme/index.ts [hunk 5]
```diff
@@ -96,6 +96,13 @@ const baseConfig: ThemeConfig = {
       colorSplit: supersetTheme.colors.grayscale.light3,
       colorText: supersetTheme.colors.grayscale.dark1,
     },
+    Modal: {
+      colorBgMask: `${supersetTheme.colors.grayscale.dark2}73`,
```
[the remark this answers is not in the document]
Comment (alexandrusoare) [reply superset-frontend/src/theme/index.ts:100]: It's for opacity, in the previous version the color was an rgba, so in order to modify the hex to match the previous rgba I added that `73`

File: superset-frontend/src/theme/index.ts [hunk 6]
```diff
@@ -96,6 +96,13 @@ const baseConfig: ThemeConfig = {
...
+      contentBg: supersetTheme.colors.grayscale.light5,
+      titleFontSize: supersetTheme.gridUnit * 4,
+      titleColor: `${supersetTheme.colors.grayscale.dark2}D9`,
```
[the remark this answers is not in the document]
Comment (alexandrusoare) [reply superset-frontend/src/theme/index.ts:103]: Same for the `D9` as well, for opacity

File: superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx (hunk 1 above)
Comment (alexandrusoare) [reply superset-frontend/src/components/ErrorMessage/ErrorAlert.tsx:74]: This a customized modal and the color of the header depends on the level of error meaning `info` , `warning` , `error`, I think this should stay as it is

File: superset-frontend/src/components/Modal/Modal.tsx (hunk 3 above)
Comment (alexandrusoare) [reply superset-frontend/src/components/Modal/Modal.tsx:125]: This is already centralized as it is configured in the main Modal component

File: superset-frontend/src/components/Modal/Modal.tsx (hunk 4 above)
Comment (alexandrusoare) [reply superset-frontend/src/components/Modal/Modal.tsx:131]: This is already centralized as it is configured in the main Modal component

File: superset-frontend/src/features/databases/DatabaseModal/styles.ts [hunk 7]
```diff
@@ -152,7 +152,6 @@ export const antDModalStyles = (theme: SupersetTheme) => css`
   }
 
   .antd5-modal-close-x .close {
-    color: ${theme.colors.grayscale.dark1};
```
Comment (rusackas) [inline superset-frontend/src/features/databases/DatabaseModal/styles.ts:155]: Always excited to see CSS disappear! :D

**Review by geido — APPROVED** [review]
Looks good to me. I see we haven't changed the Storybook. Can you confirm that all props are covered?

**Comment by alexandrusoare** [comment]
> Looks good to me. I see we haven't changed the Storybook. Can you confirm that all props are covered?

Yes, now all props are covered

**Pull request merged**